### ISSUE TITLE
Custom palette scroll speed

### ISSUE THREAD

**Issue by LilyMakesThings** [ScratchAddons/ScratchAddons#7245]
### Why this would be helpful

The palette's scrolling can be disorienting; if you're building a script really quickly it's very easy to drag out or misclick on the wrong blocks.

### How this addon works

You can select the speed you want the palette to scroll at; you can either slow it down, speed it up or remove scrolling entirely (in which case it snaps to the category you select).

**Comment by BlueGoat11** [comment]
I think that would be one of the best addons for me…On Mar 2, 2024, at 10:25 PM, LilyMakesThings ***@***.***> wrote:
Why this would be helpful
The palette's scrolling can be disorienting; if you're building a script really quickly it's very easy to drag out or misclick on the wrong blocks.
How this addon works
You can select the speed you want the palette to scroll at; you can either slow it down, speed it up or remove scrolling entirely (in which case it snaps to the category you select).
Possible alternatives
No response
Additional context
No response

**Comment by DNin01** [comment]
I'm not sure most people would care about changing the speed of the scroll animation. 🤔

The option to disable it entirely for people who are set off by it would be nice, though. 👍

But I suppose it might not hurt to have speed options. On the off-chance that someone wants to customize the scroll speed, this could be a nice thing to have around. We'll need to discuss this and decide whether we want to do that. I personally don't like to overload users with unnecessary customization options.

If we decide on just having an on/off switch for the scroll animation, we could even include this in an addon about reducing motion.

**Comment by BlueGoat11** [comment]
I agree with the customization overload ideology, but I do like the idea of a custom scroll speed, or even just a don’t scroll option. When I’m speed-scratching that little boost of a scroll speed would make a difference. For me, I can never by memory remember where each block is in each section, so just being able to scroll faster might increase productivity. Plus, it probably wouldn’t really be hard to implement…

**Comment by DNin01** [comment]
If there's enough consensus, I think this feature would be a nice addition.

**Comment by LilyMakesThings** [comment]
I've been working on a project to test this (as well as other unrelated things). I tested faster scroll speeds and slower scroll speeds. Both have their own advantages.

**Comment by BroJac5246** [comment]
I like the idea, but are there any existing addons this could be made part of? This feels too small IMO.

**Comment by BlueGoat11** [comment]
I don’t know…

**Comment by Samq64** [comment]
@BlueGoat11 If you don't have anything productive to add please just leave a comment reaction on the website instead.

**Comment by LilyMakesThings** [comment]
> I like the idea, but are there any existing addons this could be made part of? This feels too small IMO.

I think combining it with maybe being able to change how much you step up the palette when moving the scroll wheel would make it a pretty good round-about addon.

**Comment by BlueGoat11** [comment]
What if you also loop the palette as well so you can implement it into the infinite scrolling add on?

**Comment by LilyMakesThings** [comment]
Infinite scrolling is very distant from the editor.

**Comment by DNin01** [comment]
Could be part of #7251 if this is too small for its own addon.

**Comment by Secret-chest** [comment]
> Customizing the speed of the scroll animation or disabling it could be part of #7251 if it's too small for its own addon.

But the native scroll transition can't be controlled.
